api/orders/index.js

```diff
@@ -43,37 +60,86 @@ router.route('/').post(async (req, res, next) => {
   } catch (e) {
     res.status(500).json({ msg: 'Lỗi hệ thống', e })
   }
-}).put(auth, async (req, res) => {
-// @route:  PUT /api/orders
+})
+// @route:  PUT /api/orders/:id
 // @desc:   Cập nhật đơn hàng
-  const { orderId } = req.body
-  const order = await Order.findByIdAndUpdate(Mongoose.Types.ObjectId(orderId), { checking: true }).lean()
-  if (!order) {
-    return res.status(404).json({ msg: 'Không tìm thấy đơn hàng' })
-  }
+router.put('/:id', auth, async (req, res) => {
   try {
-    if (order.checking) {
-      return res.status(200).json({ msg: 'Đã cập nhật hóa đơn' })
+    const order = await Order.findById(req.params.id)
+    if (!order) {
+      return res.status(404).json({ msg: 'Không tìm thấy đơn hàng' })
     }
+    const updateOrder = await Order.findByIdAndUpdate(req.params.id, req.body, {
+      new: true
+    })
+    res
+      .status(201)
+      .json({ products: updateOrder, msg: 'Cập nhật thành công' })
   } catch (e) {
-    return res.status(500).json({ msg: 'Lỗi hệ thống' })
+    return res.status(500).json({ msg: e })
   }
 })
-
 router.route('/check').post(async (req, res, next) => {
   // @route:  POST /api/orders/check
   // @desc:   Kiểm tra tình trạng đơn hàng
   const { productId } = req.body
-  const product = await Product.findOne({ _id: Mongoose.Types.ObjectId(productId) }).lean()
-  const ordered = await Order.find({ productId: Mongoose.Types.ObjectId(productId), isCancel: false }).lean()
+  const product = await Product.findOne({
+    _id: Mongoose.Types.ObjectId(productId)
+  }).lean()
+  const ordered = await Order.find({
+    productId: Mongoose.Types.ObjectId(productId),
+    isCancel: false
+  }).lean()
   const available = product.amount - ordered.length
   if (available === 0) {
     return res.status(400).json({ msg: 'Sản phầm hết hàng', available })
   }
-  return res.status(200).json({ msg: 'Bạn có muốn tiến hành đặt hàng không?', available })
+  return res
+    .status(200)
+    .json({ msg: 'Bạn có muốn tiến hành đặt hàng không?', available })
+})
+
+router.get('/', async (req, res) => {
+  const order = await Order.aggregate([
+    {
+      $lookup: {
+        from: 'products',
+        localField: 'productId',
+        foreignField: '_id',
+        as: 'products'
+      }
+    },
+    {
+      $lookup: {
+        from: 'orders',
```

Comment:
??

Reply:
chỗ này là sao vậy a, ko được tìm kiếm trong bảng orders hả a??

Reply:
mặc định nó lấy từ bảng order ra rồi, cái lookup này nó lấy giá trị từ bảng khác với cái khoá ngoại

Reply:
à dạ a, e hiểu rồi

Reply:
remove it